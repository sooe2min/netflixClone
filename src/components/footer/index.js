import React from 'react'
import { Container, Title, Row, Column, Link, Select, Text, Copy, Break } from './styles/footer'

export default function Footer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>
}

Footer.Title = function FooterTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>
}

Footer.Row = function FooterRow({ children, ...restProps }) {
  return <Row {...restProps}>{children}</Row>
}

Footer.Column = function FooterColumn({ children, ...restProps }) {
  return <Column {...restProps}>{children}</Column>
}

Footer.Link = function FooterLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>
}

Footer.Select = function FooterSelect({ children, ...restProps }) {
  return <Select {...restProps}>{children}</Select>
}

Footer.Text = function FooterText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

Footer.Copy = function FooterCopy({ children, ...restProps }) {
  return <Copy {...restProps}>{children}</Copy>
}

Footer.Break = function FooterBreak({ children, ...restProps }) {
  return <Break {...restProps}>{children}</Break>
}
