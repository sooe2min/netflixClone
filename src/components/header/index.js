import React, { useState } from 'react'
import { Background, Container, Logo, ButtonLink, Group, Link, Profile, Search, SearchIcon, SearchInput, Picture, Dropdown, Feature, FeatureCallOut, Text, PlayButton } from './styles/header'
import { Link as ReactRouterLink } from 'react-router-dom'

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>
}

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  )
}

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>
}

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>
}

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>
}

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>
}

Header.Search = function HeaderSearch({ searchTurm, setSearchTurm, ...restProps }) {
  const [searchActive, setSearchActive] = useState(false)

  return (
    <Search {...restProps}>
      <SearchIcon onClick={() => setSearchActive(!searchActive)}>
        <img src='/images/icons/search.png' alt='search' />
      </SearchIcon>
      <SearchInput
        placeholder='Search films and series'
        value={searchTurm}
        onChange={({ target }) => setSearchTurm(target.value)}
        active={searchActive}
      />
    </Search >
  )
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />
}

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>
}


Header.FeatureCallOut = function HeaderFeatureCallOut({ children, ...restProps }) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>
}

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>
}
