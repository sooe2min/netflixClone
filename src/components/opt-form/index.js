import React from 'react'
import { Container, Text, Input, Button, Break } from './styles/opt-form'

export default function OptForm({ children, ...restProps }) {
  return (
    <Container {...restProps}>
      {children}
    </Container>
  )
}

OptForm.Text = function OptFormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

OptForm.Input = function OptFromInput({ children, ...restProps }) {
  return <Input {...restProps}>{children}</Input>
}

OptForm.Button = function OptFormButton({ children, ...restProps }) {
  return (
    <Button {...restProps}>
      {children} <img src='/images/icons/chevron-right.png' alt='Get Started' />
    </Button>
  )
}

OptForm.Break = function OptFormBreak({ children, ...restProps }) {
  return <Break {...restProps} />
}