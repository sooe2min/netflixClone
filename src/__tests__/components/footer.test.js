import React from 'react'
import { render } from '@testing-library/react'
import { Footer } from '../../components'

describe('<Footer />', () => {
  it('renders the <Footer /> with populated data', () => {
    const { container, getByText } = render(
      <Footer>
        <Footer.Title>Questions? Call 00-308-321-0161</Footer.Title>
        <Footer.Break />
        <Footer.Row>
          <Footer.Column>
            <Footer.Link href="#">FAQ</Footer.Link>
            <Footer.Link href="#">Investor Relations</Footer.Link>
            <Footer.Link href="#">Privacy</Footer.Link>
            <Footer.Link href="#">Speed Test</Footer.Link>
          </Footer.Column>
          <Footer.Column>
            <Footer.Link href="#">Help Center</Footer.Link>
            <Footer.Link href="#">Jobs</Footer.Link>
            <Footer.Link href="#">Cookie Preferences</Footer.Link>
            <Footer.Link href="#">Legal Notices</Footer.Link>
          </Footer.Column>
          <Footer.Column>
            <Footer.Link href="#">Account</Footer.Link>
            <Footer.Link href="#">Ways to Watch</Footer.Link>
            <Footer.Link href="#">Corporate Information</Footer.Link>
            <Footer.Link href="#">Netflix Originals</Footer.Link>
          </Footer.Column>
          <Footer.Column>
            <Footer.Link href="#">Media Center</Footer.Link>
            <Footer.Link href="#">Terms of Use</Footer.Link>
            <Footer.Link href="#">Contact Us</Footer.Link>
          </Footer.Column>
        </Footer.Row>
        <Footer.Select>
          <option>English</option>
          <option>한국어</option>
        </Footer.Select>
        <Footer.Text>Netflix South Korea</Footer.Text>
        <Footer.Break />
        <Footer.Copy>
          Netflix Services Korea Ltd. E-Commerce Registration Number: Je 2018-Seoul Jong-ro-0426 Ho. Phone: 00-308-321-0161 <br />
        Representative: Reginald Shawn Thompson <br />
        Email: korea@netflix.com <br />
        Address: 20F, Tower A, Centropolis Building 26, Ujeongguk-ro, Jongno-gu, Seoul, 03161 Republic of Korea <br />
        Hosted by: Amazon Web Services Inc. <br />
          <a href="https://www.ftc.go.kr/www/bizCommView.do?key=232&apv_perm_no=2018300016930200431&pageUnit=10&searchCnd=bup_nm&searchKrwd=%EB%84%B7%ED%94%8C%EB%A6%AD%EC%8A%A4&pageIndex=1">Link to KFTC website</a>
        </Footer.Copy>
      </Footer >
    )

    expect(getByText('Questions? Call 00-308-321-0161')).toBeTruthy()
    expect(getByText('FAQ')).toBeTruthy()
    expect(getByText('Help Center')).toBeTruthy()
    expect(getByText('Netflix Originals')).toBeTruthy()
    expect(getByText('Contact Us')).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})