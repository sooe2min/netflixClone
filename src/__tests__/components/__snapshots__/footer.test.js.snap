// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer /> with populated data 1`] = `
<div
  class="footer__Container-qu6i6-0 eVeZpC"
>
  <div
    class="footer__Title-qu6i6-1 eaYck"
  >
    Questions? Call 00-308-321-0161
  </div>
  <p
    class="footer__Break-qu6i6-8 gmGiXt"
  />
  <div
    class="footer__Row-qu6i6-2 dQGCME"
  >
    <div
      class="footer__Column-qu6i6-3 jtkbGA"
    >
      <a
        class="footer__Link-qu6i6-4 gIdRCz"
        href="#"
      >
        FAQ
      </a>
      <a
        class="footer__Link-qu6i6-4 gIdRCz"
        href="#"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-qu6i6-4 gIdRCz"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-qu6i6-4 gIdRCz"
        href="#"
      >
        Speed Test
      </a>
    </div>
    <div
      class="footer__Column-qu6i6-3 jtkbGA"
    >
      <a
        class="footer__Link-qu6i6-4 gIdRCz"
        href="#"
      >
        Help Center
      </a>
      <a
        class="footer__Link-qu6i6-4 gIdRCz"
        href="#"
      >
        Jobs
      </a>
      <a
        class="footer__Link-qu6i6-4 gIdRCz"
        href="#"
      >
        Cookie Preferences
      </a>
      <a
        class="footer__Link-qu6i6-4 gIdRCz"
        href="#"
      >
        Legal Notices
      </a>
    </div>
    <div
      class="footer__Column-qu6i6-3 jtkbGA"
    >
      <a
        class="footer__Link-qu6i6-4 gIdRCz"
        href="#"
      >
        Account
      </a>
      <a
        class="footer__Link-qu6i6-4 gIdRCz"
        href="#"
      >
        Ways to Watch
      </a>
      <a
        class="footer__Link-qu6i6-4 gIdRCz"
        href="#"
      >
        Corporate Information
      </a>
      <a
        class="footer__Link-qu6i6-4 gIdRCz"
        href="#"
      >
        Netflix Originals
      </a>
    </div>
    <div
      class="footer__Column-qu6i6-3 jtkbGA"
    >
      <a
        class="footer__Link-qu6i6-4 gIdRCz"
        href="#"
      >
        Media Center
      </a>
      <a
        class="footer__Link-qu6i6-4 gIdRCz"
        href="#"
      >
        Terms of Use
      </a>
      <a
        class="footer__Link-qu6i6-4 gIdRCz"
        href="#"
      >
        Contact Us
      </a>
    </div>
  </div>
  <select
    class="footer__Select-qu6i6-5 dtflxu"
  >
    <option>
      English
    </option>
    <option>
      한국어
    </option>
  </select>
  <div
    class="footer__Text-qu6i6-6 gbMczL"
  >
    Netflix South Korea
  </div>
  <p
    class="footer__Break-qu6i6-8 gmGiXt"
  />
  <div
    class="footer__Copy-qu6i6-7 cdGFJz"
  >
    Netflix Services Korea Ltd. E-Commerce Registration Number: Je 2018-Seoul Jong-ro-0426 Ho. Phone: 00-308-321-0161 
    <br />
    Representative: Reginald Shawn Thompson 
    <br />
    Email: korea@netflix.com 
    <br />
    Address: 20F, Tower A, Centropolis Building 26, Ujeongguk-ro, Jongno-gu, Seoul, 03161 Republic of Korea 
    <br />
    Hosted by: Amazon Web Services Inc. 
    <br />
    <a
      href="https://www.ftc.go.kr/www/bizCommView.do?key=232&apv_perm_no=2018300016930200431&pageUnit=10&searchCnd=bup_nm&searchKrwd=%EB%84%B7%ED%94%8C%EB%A6%AD%EC%8A%A4&pageIndex=1"
    >
      Link to KFTC website
    </a>
  </div>
</div>
`;
